### PR TITLE
Address potential causes for "Your search did not match any code" issue when clicking a language in the stats bar

### PR DESCRIPTION
**Pull request by DecimalTurn**
Following up on: https://github.com/github-linguist/linguist/discussions/6408
Feedback is welcome

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -18,6 +18,14 @@ If the language stats bar is reporting a language that you don't expect:
...
 
+## When I click on a language present in the language stats bar, I get a message saying "Your search did not match any code"
+
+There are a few reasons that could lead to this outcome:
+
+1. If you are making an override via the `linguist-language` attribute, it will not be taken into account in GitHub's search results since GitHub Search relies on [go-enry](https://github.com/go-enry/go-enry) which doesn't support overrides at the moment ([More info](https://github.com/src-d/enry/issues/18)).
+2. It could be due to the fact that go-enry isn't using the latest version of Linguist which means that files could be detected differently in search compared to what Linguist detects for the status bar (see also the note at the end of [this section](#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes)).
+3. Rarely, it can be that the files are associated to a language that is part of a group, this means that they will be counted as the parent language in the language stats bar, but as the actual language in search. For instance, a file ending with `.f90`, it is considered to be a "Fortran" in the stats bar, but "Fortran Free Form" in search.
+
```
Comment (lildude) [inline docs/troubleshooting.md:28]: There’s also a fourth explanation: the [code search limitations](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations). I thought we had a reference to these somewhere but I can’t find it now (AFK on mobile only) so I think we should mention them here too.

Comment (DecimalTurn) [reply docs/troubleshooting.md:29]: Good to know! I've added that to the list.

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -69,5 +78,5 @@ All syntax highlighting grammars will also be updated in all major and minor rel
 Grammars will only be updated in patch releases if the patch release is specifically for that language and it requires a grammar update to address the issue.
 
 Note: New languages will not appear in GitHub's search results for some time after the pull request has been merged and the new Linguist release deployed to GitHub.com.
-This is because GitHub's search uses [go-enry](https://github.com/go-enry/go-enry) for language detection which tends to lag behind Linguist by a few weeks to months.
+This is because GitHub's search uses go-enry for language detection which tends to lag behind Linguist by a few weeks to months.
```
Comment (lildude) [inline docs/troubleshooting.md:81]: Please keep this link. Peeps are lazy and may skip straight to this and then wonder WTF `go-enry` is.

Comment (DecimalTurn) [reply docs/troubleshooting.md:81]: Sure thing!

**Review by lildude — APPROVED** [review]

**Comment by DecimalTurn** [comment]
Looks like the github-merge-queue tried to merge a [squashed commit](https://github.com/github-linguist/linguist/commit/e4818cc) of those changes but the fact that it doesn't belong to a branch made it fail. 🤔

**Comment by lildude** [comment]
Blast!! Looks like a legit failure, probably due to a gem update or change somewhere. I’ll take a look tomorrow.

**Comment by lildude** [comment]
Yup, it's a gem compat issue: https://github.com/freerange/mocha/issues/614. Fixed in https://github.com/github-linguist/linguist/pull/6499

**Pull request merged**